src/services/contract_manager.js

```diff
@@ -7,22 +7,89 @@ This Source Code Form is subject to the terms of the Mozilla Public License, v.
 This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
 */
 
-import {loadContract, deployContract} from '../utils/web3_tools';
-import BundleRegistry from '../../contracts/BundleRegistry.json';
+import {getDefaultAddress, loadContract} from '../utils/web3_tools';
+import ChallengesJson from '../../contracts/Challenges.json';
+import ContextJson from '../../contracts/Context.json';
+import FeesJson from '../../contracts/Fees.json';
+import HeadJson from '../../contracts/Head.json';
+import KycWhitelistJson from '../../contracts/KycWhitelist.json';
+import PayoutsJson from '../../contracts/Payouts.json';
+import RolesJson from '../../contracts/Roles.json';
+import ShelteringTransfersJson from '../../contracts/ShelteringTransfers.json';
+import UploadsJson from '../../contracts/Uploads.json';
+
 
 export default class ContractManager {
-  static loadBundleRegistryContract(web3, bundleRegistryContractAddress) {
-    let result;
-    if (bundleRegistryContractAddress) {
-      result = loadContract(web3, BundleRegistry.abi, bundleRegistryContractAddress);
+  constructor(web3, headContractAddress) {
+    this.web3 = web3;
+    if (headContractAddress) {
+      this.head = loadContract(web3, HeadJson.abi, headContractAddress);
     } else {
-      throw new Error('bundle proof registry contract address is not configured');
+      throw new Error('Head contract address is not configured');
+    }
+  }
+
+  async getContractAddress(contractName) {
+    const availableContracts = [
+      'kycWhitelist',
+      'roles',
+      'fees',
+      'challenges',
+      'payouts',
+      'shelteringTransfers',
+      'uploads'
+    ];
+
+    if (!availableContracts.includes(contractName)) {
+      throw new Error('Requested contract does not exist');
     }
-    return result;
+
+    const context = await this.getContext();
+    const from  = getDefaultAddress(this.web3);
+    return context.methods[`${contractName}()`]().call({from});
+  }
+
+  async getContext() {
+    const from  = getDefaultAddress(this.web3);
+    const contextAddress = await this.head
+      .methods
+      .context()
+      .call({from});
+    return loadContract(this.web3, ContextJson.abi, contextAddress);
+  }
+
+  async getKycWhitelistContract() {
+    const contractAddress = await this.getContractAddress('kycWhitelist');
+    return loadContract(this.web3, KycWhitelistJson.abi, contractAddress);
```

Comment:
If you create a map: 
```
{
   "kycWhiteList": KycWhitelistJson,
   ...
}
```

you can simplify and deduplicate all the get methods to just `return contractByKey('kycWhiteList')`

Reply:
done 👍